clients/Application/src/app/views/auth/auth-configuration.service.ts

```diff
@@ -14,167 +14,82 @@
-
-    this.http.get<OpenIdConfigParams>(stsServer+"/.well-known/openid-configuration")
-    .subscribe((openIdConfigParams) => {
-      const url = openIdConfigParams.authorization_endpoint.replace("oauth2/authorize", "");
-      window.location.href = url+`logout?client_id=${clientId}&logout_uri=${logoutUrl}`;
-    })
-    return;
-  }
+    private route: ActivatedRoute,
+    private router: Router
+  ) {}
 
   public setTenantConfig(tenantName: string): Promise<any> {
-    const url = `${environment.regApiGatewayUrl}/tenant/init/`+tenantName;
+    const url = `${environment.regApiGatewayUrl}/tenant/init/` + tenantName;
     this.params$ = this.http.get<ConfigParams>(url);
-
-    const setup$ =this.params$.pipe(
+    const setup$ = this.params$.pipe(
       map((val) => {
-
-        localStorage.setItem('userPoolId',val.userPoolId);
-        localStorage.setItem('appClientId',val.appClientId);
-        localStorage.setItem('apiGatewayUrl',val.apiGatewayUrl);
-        this.setOidcConfig();
-        return "success";
-
+        // remove trailing slash (/) if present
+        val.apiGatewayUrl = val.apiGatewayUrl.replace(/\/$/, '');
+        localStorage.setItem('userPoolId', val.userPoolId);
+        localStorage.setItem('tenantName', tenantName);
+        localStorage.setItem('appClientId', val.appClientId);
+        localStorage.setItem('apiGatewayUrl', val.apiGatewayUrl);
+        return 'success';
       }),
       catchError((error) => {
-        console.log(error)
+        console.log('Error setting tenant config: ', error);
         return throwError(error);
       })
     );
 
-
-      return setup$.toPromise();
-
+    return setup$.toPromise();
   }
 
-  public setOidcConfig() {
-    let appClientId = localStorage.getItem('appClientId');
-    let userPoolId = localStorage.getItem('userPoolId');
-
-    if(this.isValid(appClientId)
-    && this.isValid(userPoolId)) {
-      let region = userPoolId.split('_')[0];
-      console.log(window.location.origin);
-
-          this.oidcConfigService.withConfig({
-            stsServer: `https://cognito-idp.${region}.amazonaws.com/${userPoolId}`,
-            redirectUrl: `${window.location.origin}/`,
-            clientId: appClientId,
-            responseType: 'code',
-            scope: 'openid profile email',
-            postLogoutRedirectUri: `${window.location.origin}/signout`,
-            silentRenew: false, // customConfig.useSilentRefresh,
-            postLoginRoute: '',
-            forbiddenRoute: '/forbidden',
-            unauthorizedRoute: '/unauthorized',
-            logLevel: LogLevel.Debug,
-            maxIdTokenIatOffsetAllowedInSeconds: 10,
-            historyCleanupOff: true,
-            useRefreshToken: true,
-            autoUserinfo: false,
-
-          });
-
-        }
-
+  configureAmplifyAuth(): boolean {
+    try {
+      const userPoolId = localStorage.getItem('userPoolId');
+      const appClientId = localStorage.getItem('appClientId');
```

Comment:
If these values are not present, does it through an exception?

Reply:
Add a comment to ensure that `Amplify.configure()` will fail if not all values are present.

Reply:
Added a check in lines 74 and 75 to ensure that we never run `Amplify.configure()` with empty values:
```
      if (!userPoolId || !appClientId) {
        return false;
      }
```